### ISSUE TITLE
Key created using the above steps is not valid for signing a UWP application.

### ISSUE THREAD

**Issue by Tohaker** [MicrosoftDocs/msix-docs#41]
I followed the steps exactly, creating a pfx key for use in my Azure Pipeline as well as for local builds. Whenever it runs on the Pipeline, I get the following error: 
"##[error]C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Microsoft\VisualStudio\v16.0\AppxPackage\Microsoft.AppXPackage.Targets(4432,5): Error APPX0107: The certificate specified is not valid for signing. For more information about valid certificates, see http://go.microsoft.com/fwlink/?LinkID=241478."

When trying to add it my local project in the "Signing" tab of the project properties, it gives me the error: "Error importing key. Invalid provider type specifed."

Where have I gone wrong?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 8e95ba48-8d07-d390-d49b-801d70730bae
* Version Independent ID: 047d2736-f14e-610c-7396-89d2eae0f9cc
* Content: [Create a certificate for package signing - MSIX](https://docs.microsoft.com/en-us/windows/msix/package/create-certificate-package-signing#feedback)
* Content Source: [msix-src/package/create-certificate-package-signing.md](https://github.com/microsoftdocs/msix-docs/blob/master/msix-src/package/create-certificate-package-signing.md)
* Product: **msix**
* GitHub Login: @dianmsft
* Microsoft Alias: **diahar**

**Comment by dianmsft** [comment]
To sign your MSIX pacakge the  cert/keys must be SHA256 digest, must be RSA 2048 keys. Also important thing to do is to add the cert to the local cert store, otherwise it will not be trusted on the dev's machine that is being sideloaded.

**Comment by Jahanaray** [comment]
It's ridicules that in newer version of VS developers should do more! and have less convenient in developing! We need the autosiging feature again!

**Comment by MathiasMagnus** [comment]
I too followed these steps in my attempt to sign a package created by the MSIX Packaging Tool. I created the pacakge in a VM without signing, copied it over to my devbox, and wanted to sign with a freshly generated self-sign key. The key was created via

```powershell
New-SelfSignedCertificate -Type Custom -Subject "CN=Something" -KeyUsage DigitalSignature -FriendlyName "MySomething" -CertStoreLocation "Cert:\CurrentUser\My" -TextExtension @("2.5.29.37={text}1.3.6.1.5.5.7.3.3", "2.5.29.19={text}") -KeyAlgorithm RSA -KeyLength 2048
```

it was exported to pfx with a password, I used that password to import it to both Cert:\CurrentUser\My AND under Cert:\LocalMachine\My using `certmgr` but nothing! Must it really be that hard? I burned more short of 2 hours trying to figure out package signing.
